bin/metrics-vsphere.rb

```diff
@@ -0,0 +1,355 @@
+#! /usr/bin/env ruby
+#
+#   metrics-vsphere
+#
+# DESCRIPTION:
+#
+# OUTPUT:
+#   plain text, metric data, etc
+#
+# PLATFORMS:
+#   Linux, Windows, BSD, Solaris, etc
+#
+# DEPENDENCIES:
+#   gem: sensu-plugin
+#   gem: rbvmomi
+#
+# USAGE:
+#
+# NOTES:
+#
+# LICENSE:
+#   Yuri Zubov  <yury.zubau@gmail.com>
+#   Released under the same terms as Sensu (the MIT license); see LICENSE
+#   for details.
+#
+
+require 'sensu-plugin/metric/cli'
+require 'rbvmomi'
+
+#
+# VSphere Graphite
+#
+class VsphereGraphite < Sensu::Plugin::Metric::CLI::Graphite
+  option :host,
+         description: 'ESX or ESXi hostname',
+         short: '-H HOST',
+         long: '--host HOST'
+
+  option :user,
+         description: 'Username to connect with',
+         short: '-u USER',
+         long: '--user USER',
+         default: 'root'
+
+  option :password,
```

Comment:
please leave a comment about the dangers of not using token substitution especially since this likely seems like a root account on a hypervisor.

https://sensuapp.org/docs/latest/reference/checks.html#check-token-substitution